octree/include/pcl/octree/impl/octree_iterator.hpp

```diff
@@ -281,6 +281,65 @@ namespace pcl
 
       return (*this);
     }
+
+    //////////////////////////////////////////////////////////////////////////////////////////////
+    template<typename OctreeT>
+    OctreeFixedDepthIterator<OctreeT>::OctreeFixedDepthIterator (unsigned int depth_arg) :
+        OctreeBreadthFirstIterator<OctreeT> (depth_arg), depth_(depth_arg)
+    {
+    }
+
+    //////////////////////////////////////////////////////////////////////////////////////////////
+    template<typename OctreeT>
+    OctreeFixedDepthIterator<OctreeT>::OctreeFixedDepthIterator (OctreeT* octree_arg, unsigned int depth_arg) :
+        OctreeBreadthFirstIterator<OctreeT> (octree_arg, depth_arg), depth_(depth_arg)
+    {
+      this->reset (depth_arg);
+    }
+
+    //////////////////////////////////////////////////////////////////////////////////////////////
+    template<typename OctreeT>
+    void OctreeFixedDepthIterator<OctreeT>::reset ()
+    {
+      OctreeBreadthFirstIterator<OctreeT>::reset ();
+
+      // Set the current_state_ to the right depth
+      while (this->current_state_ && (this->getCurrentOctreeDepth () != depth_))
+      {
+        OctreeBreadthFirstIterator<OctreeT>::operator++ ();
+      }
+    }
+
+    //////////////////////////////////////////////////////////////////////////////////////////////
+    template<typename OctreeT>
+    void OctreeFixedDepthIterator<OctreeT>::reset (unsigned int depth)
+    {
+      if ((0 <= depth) && (depth <= this->octree_->getTreeDepth ()))
+      {
+        this->depth_ = depth;
+      }
+      else
+      {
+        this->depth_ = 0;
+      }
+
+      this->reset ();
+    }
```

Comment:
Based on my previous suggestion you can delete this guy.

Reply:
Ok, now there is only one `reset()` method:
```cpp
/** \brief Reset the iterator to the first node at the depth given as parameter
*/
void
reset (unsigned int depth = 0u);
```

Reply:
In recent commits, I introduced a `reset()` method without argument. It allows the iterator to keep the same internal `fixed_depth_` if someone calls the `reset()` method without argument. I think it's the right behavior. It comes clear for me when I checked the unit test with `OctreeBreadthFirstIterator`.